Firmware/Marlin_main.cpp

```diff
@@ -1754,6 +1754,10 @@ void stack_error() {
     crash_and_burn(dump_crash_reason::stack_error);
 }
 
+void pullup_error(bool fromTempISR) {
+    crash_and_burn(fromTempISR ? dump_crash_reason::bad_pullup_temp_isr : dump_crash_reason::bad_pullup_step_isr);
```

Comment:
please do the `#ifdef DEBUG_PULLUP_CRASH` in this function like this:

```
void pullup_error(bool fromTempISR) {
#ifdef DEBUG_PULLUP_CRASH
    crash_and_burn(fromTempISR ? dump_crash_reason.....
#else
    pull_up.disable();
#endif
}
```

This way you can avoid the #ifdefs at the calling places and you'd be ready for the production safety feature which turns off the pull-up if encountered.

Reply:
I don't like this approach. It would make no sense to constantly check if the pullups got enabled and then disable them. We can just disable them right away. Much quicker like this

Reply:
ok, then please add the "disable pull-up" into the #else branches at the correct spots in this PR